### DISCUSSION TITLE
How to add roles to new user after user registration by programming,please?

### DISCUSSION THREAD

**Discussion by effapp** [OrchardCMS/OrchardCore#16460] [Q&A] [answered]
How to add roles to new user after user registration by programming in the custom module,please?

**Comment by Piedone** [comment]
You can implement `IRegistrationFormEvents.RegisteredAsync` and then do what [the Roles module does](https://github.com/OrchardCMS/OrchardCore/blob/27142a0626e7e71ac27e750f82b01d93e365405f/src/OrchardCore.Modules/OrchardCore.Roles/Controllers/AdminController.cs#L22).

**Comment by effapp** [comment]
Hi @Piedone ,
     This [the Roles module does](https://github.com/OrchardCMS/OrchardCore/blob/27142a0626e7e71ac27e750f82b01d93e365405f/src/OrchardCore.Modules/OrchardCore.Roles/Controllers/AdminController.cs#L22) don't include codes to add roles to user.

**Comment by agriffard** [comment, marked as answer]
https://github.com/OrchardCMS/OrchardCore/blob/8d403cda32800723a62d4d100474293994fd0316/src/OrchardCore.Modules/OrchardCore.Users/Workflows/Activities/AssignUserRoleTask.cs#L65

**Comment by effapp** [comment]
Thanks.
